### PR TITLE
LNX-126 Add displaying tree object

### PR DESCRIPTION
**Pull request by Krystian030**
- Add chop action
- Add create object action
- Add remove object action
- Add populate button

### CODE REVIEW

File: ui/Populate.gd [hunk 1]
```diff
@@ -0,0 +1,13 @@
+extends Node
+
+@onready var post_populate_endpoint_url = "http://127.0.0.1:8555/populate"
```
Comment (filip-szweda) [inline ui/Populate.gd:3]: `SERVER_ADDRESS` + "populate"

**Review by filip-szweda — CHANGES_REQUESTED** [review]

File: ui/Populate.gd (hunk 1 above)
Comment (Krystian030) [reply ui/Populate.gd]: Done

File: ui/CodeEditor.gd [hunk 2]
```diff
@@ -29,6 +29,6 @@ func _ready():
 	syntax_highlighter.set_number_color(number_color)
 	syntax_highlighter.set_symbol_color(symbol_color)
 	syntax_highlighter.set_member_variable_color(member_variable_color)
-	
+	self.text = "move(Vector(1,0))\nchop(Vector(1,0))"
```
[the remark this answers is not in the document]
Comment (Krystian030) [reply ui/CodeEditor.gd]: Done

File: ui/CodeEditor.gd [hunk 3]
```diff
@@ -1,6 +1,6 @@
 extends CodeEdit
 
-const keywords = ['dupa', 'and', 'assert', 'break', 'class', 'continue', 'def', 'del', 'elif', 'else',
```
[the remark this answers is not in the document]
Comment (Krystian030) [reply ui/CodeEditor.gd:3]: Done

File: scene/world_updater/DeltasApplier.gd [hunk 4]
```diff
@@ -11,8 +12,12 @@ func apply_deltas(deltas_json):
 		json.parse(delta_json)
 		var delta = json.get_data()
 		var entity = entity_deserializer.deserialize(delta)
+		if not "_object_id" in entity:
```
[the remark this answers is not in the document]
Comment (Krystian030) [reply scene/world_updater/DeltasApplier.gd]: Done

File: scene/world_updater/DeltasApplier.gd [hunk 5]
```diff
@@ -11,8 +12,12 @@ func apply_deltas(deltas_json):
...
+			self.global_action_queue.add_child(entity)
+			return
 		if !entity:
 			print("[ERROR] Could not deserialize Entity")
 		if entity is LynxAction:
+			print(entity.name)
```
[the remark this answers is not in the document]
Comment (Krystian030) [reply scene/world_updater/DeltasApplier.gd]: Done

File: entity/object/agent.tscn [hunk 6]
```diff
@@ -223,7 +267,7 @@ script = ExtResource("1_vcg4i")
 
 [node name="AnimatedSprite2D" type="AnimatedSprite2D" parent="."]
 sprite_frames = SubResource("SpriteFrames_gslp2")
-animation = &"walk_down"
+animation = &"chop_down"
```
[the remark this answers is not in the document]
Comment (Krystian030) [reply entity/object/agent.tscn]: Done

File: ui/Populate.gd [hunk 7]
```diff
@@ -0,0 +1,13 @@
+extends Node
```
Comment (alpinus4) [inline ui/Populate.gd:1]: Please make populate button work on the same principles as creating agent button.
The UI node 
image
should catch pressing populate button and emit custom signal.
image
similarly as above.

Then in the scene you could create 'Populator'
or similarly named node.
image
that would handle the logic of actually sending populate request.
Thanks to that we keep logic outside of UI.

File: scene/world_updater/DeltasApplier.gd [hunk 8]
```diff
@@ -2,17 +2,29 @@ extends Node
 
 @onready var entity_deserializer = get_node("../EntityDeserializer")
 @onready var objects_container = get_owner().objects_container
+@onready var global_action_queue = get_node("../../GlobalActionQueue/Queue")
 var json = JSON.new()
 
 func apply_deltas(deltas_json):
 	json.parse(deltas_json)
 	var deltas = json.get_data()
+	print(">>>>>>>>>>>>>>>>>>> DELTAS <<<<<<<<<<<<<<<<<")
```
Comment (alpinus4) [inline scene/world_updater/DeltasApplier.gd:11]: please remove prints here

File: scene/world_updater/DeltasApplier.gd [hunk 9]
```diff
@@ -2,17 +2,29 @@ extends Node
 
 @onready var entity_deserializer = get_node("../EntityDeserializer")
 @onready var objects_container = get_owner().objects_container
+@onready var global_action_queue = get_node("../../GlobalActionQueue/Queue")
```
Comment (alpinus4) [inline scene/world_updater/DeltasApplier.gd:5]: lets use get_owner() where possible
`get_owner().get_node("GlobalActionQueue/Queue")`

File: entity/object/tree/Tree.gd [hunk 10]
```diff
@@ -0,0 +1,31 @@
+extends LynxObject
+
+func init(_position, _id, _owner, _tick):
+	self._position = _position
+	self._id = _id
+	self._owner = _owner
+
+# payload string used specifically for POST requests to scene-host
```
Comment (alpinus4) [inline entity/object/tree/Tree.gd:8]: both functions 
`func to_payload_string(dict: Dictionary):`
and 
`func serialize():`
are not needed, please remove them

File: entity/action/remove_object/RemoveObject.gd [hunk 11]
```diff
@@ -0,0 +1,11 @@
+extends LynxAction
+
+var _object_id = int()
+@onready var objects_container = get_node("../../../../ObjectsContainer")
```
Comment (alpinus4) [inline entity/action/remove_object/RemoveObject.gd:4]: The tree is as follows here
ObjectsContainer
-> Object
-> -> Queue
->->-> RemoveObjectAction

so change it to
```gdscript
@onready var objects_container = get_parent().get_parent().get_parent()
```

File: entity/action/create_object/CreateObject.gd [hunk 12]
```diff
@@ -0,0 +1,23 @@
+extends LynxAction
+
+var _serialized_object = int()
+@onready var objects_container = get_node("../../../../ObjectsContainer")
```
Comment (alpinus4) [inline entity/action/create_object/CreateObject.gd:4]: This is inside global action queue. Thereof to access ObjectContainer we should get it from WorldUpdater.
```gdscript
@onready var objects_container = get_parent().get_parent().get_parent().objects_container
```

**Review by alpinus4 — CHANGES_REQUESTED** [review]

File: scene/world_updater/DeltasApplier.gd (hunk 8 above)
Comment (Krystian030) [reply scene/world_updater/DeltasApplier.gd:11]: done

File: scene/world_updater/DeltasApplier.gd (hunk 9 above)
Comment (Krystian030) [reply scene/world_updater/DeltasApplier.gd:5]: Done

File: entity/object/tree/Tree.gd (hunk 10 above)
Comment (Krystian030) [reply entity/object/tree/Tree.gd:8]: done

File: entity/action/create_object/CreateObject.gd (hunk 12 above)
Comment (Krystian030) [reply entity/action/create_object/CreateObject.gd:4]: done

File: entity/action/remove_object/RemoveObject.gd (hunk 11 above)
Comment (Krystian030) [reply entity/action/remove_object/RemoveObject.gd:4]: done

**Review by alpinus4 — APPROVED** [review]
Looks good

**Review by filip-szweda — APPROVED** [review]

**Pull request merged**